### PR TITLE
EVG-2022 Add pprof endpoint to server

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: cli/service.go [hunk 1]
```diff
@@ -28,8 +29,6 @@ var (
 
 type ServiceWebCommand struct {
 	ConfigPath string `long:"conf" default:"/etc/mci_settings.yml" description:"path to the service configuration file"`
-	APIService bool   `long:"api" description:"run the API service (default port: 8080)"`
-	UIService  bool   `long:"ui" description:"run the UI service (default port: 9090)"`
```
Comment (tychoish) [inline cli/service.go]: I think we need to get the service files edited to remove use of these flags before we remove them, or else the services will fail to start.

File: cli/service.go [hunk 2]
```diff
@@ -163,3 +174,24 @@ func getHandlerUI(settings *evergreen.Settings) (http.Handler, error) {
 
 	return n, nil
 }
+
+func getHandlerPprof(settings *evergreen.Settings) http.Handler {
```
Comment (tychoish) [inline cli/service.go]: move to service

File: cli/service.go [hunk 3]
```diff
@@ -88,9 +89,19 @@ func (c *ServiceWebCommand) Execute(_ []string) error {
 		err = service.RunGracefully(settings.Ui.HttpListenAddr, requestTimeout, uiHandler)
 		close(uiWait)
 	}()
+	pprofWait := make(chan struct{})
+	if settings.PprofPort != "" {
```
Comment (tychoish) [inline cli/service.go]: do we also want this on the runner?

File: util/pprof.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
+package util
```
Comment (tychoish) [inline util/pprof.go]: maybe move this to service and maybe you can make all the functions private?

File: cli/service.go (hunk 2 above)
Comment (bsamek) [reply cli/service.go]: done

File: cli/runner.go [hunk 5]
```diff
@@ -77,6 +78,13 @@ func (c *ServiceRunnerCommand) Execute(_ []string) error {
 		return runProcessByName(ctx, c.Single, settings)
 	}
 
+	pprofHandler := service.GetHandlerPprof(settings)
+	if settings.PprofPort != "" {
+		go func() {
+			err = service.RunGracefully(settings.PprofPort, requestTimeout, pprofHandler)
```
Comment (tychoish) [inline cli/runner.go]: grip.Alert(service.RunGracefully()) 

(we're not doing anything useful with this error message.

**Pull request merged**